app/src/test/java/com/x8bit/bitwarden/ui/platform/feature/settings/exportvault/ExportVaultViewModelTest.kt

```diff
@@ -479,19 +480,60 @@ class ExportVaultViewModelTest : BaseViewModelTest() {
         )
     }
 
+    @Suppress("MaxLineLength")
     @Test
-    fun `SendCodeClick should call requestOneTimePasscode`() {
+    fun `SendCodeClick should call requestOneTimePasscode and update dialog state to sending then back to null when request completes`() =
+        runTest {
         val viewModel = createViewModel()
         coEvery { authRepository.requestOneTimePasscode() } returns RequestOtpResult.Success
-        viewModel.trySendAction(ExportVaultAction.SendCodeClick)
-
-        assertEquals(
-            DEFAULT_STATE,
-            viewModel.stateFlow.value,
-        )
+            viewModel.stateFlow.test {
+                assertEquals(DEFAULT_STATE, awaitItem())
+                viewModel.trySendAction(ExportVaultAction.SendCodeClick)
+                assertEquals(
+                    DEFAULT_STATE.copy(
+                        dialogState = ExportVaultState.DialogState.Loading(
+                            message = R.string.sending.asText(),
+                        ),
+                    ),
+                    awaitItem(),
+                )
+                assertEquals(DEFAULT_STATE, awaitItem())
+            }
         coVerify { authRepository.requestOneTimePasscode() }
     }
 
+    @Test
+    fun `On successful requestOneTimePasscode should send success toast`() = runTest {
+        val viewModel = createViewModel()
+        coEvery { authRepository.requestOneTimePasscode() } returns RequestOtpResult.Success
+        viewModel.eventFlow.test {
+            viewModel.trySendAction(ExportVaultAction.SendCodeClick)
+            assertEquals(
+                ExportVaultEvent.ShowToast(
+                    message = R.string.code_sent.asText(),
+                ),
+                awaitItem(),
+            )
+        }
+    }
+
+    @Test
+    fun `On failed requestOneTimePasscode should send success toast`() = runTest {
+        val viewModel = createViewModel()
+        coEvery {
+            authRepository.requestOneTimePasscode()
+        } returns RequestOtpResult.Error(message = null)
+        viewModel.eventFlow.test {
```

Comment:
Can we do a combined `stateEventFlow` so we can show the changes in the dialog state. Same for the other tests.